### DISCUSSION TITLE
✅ 🔗 Host Google Fonts Locally (Related to #4544)

### DISCUSSION THREAD

**Discussion by ProfessorTechie** [elementor/elementor#19966] [Feature Request]
I wish to ask that a option be included to download and host locally, the fonts your are utilizing on your site.

**To explain:** 

While I am aware of the fact that I can download the google fonts I desire to use. Then upload them to my host and choose them in elementor as a custom font. 

What I'm requesting is that if I choose a Google font(s) when making my website with Elementor, I'd like a button that downloads the font(s) to a local file(s) in WordPress. Then have Elementor select those local files as the font used. 

Opposed to haveing it linked to download from the Google repository for each visitor.

Having the font(s) externally linked slows down the page load and is marked as bad SEO in the vast majority of SEO checkers. Hosting the font(s) on your own domain as local files appears to be a much better way and enhances your SEO.

Additionally since Google fonts are, for the most part, open sourced licensed, the licenses allow redistribution. Which should make this request have no legal issues in implementing.

Though, I suspect, for some there is may be a citation requirement for some fonts that can be applied in meta to cover any legal citation licensing requirements.

Thank you for your time, and everything you guys do at Elementor.

Best Regards,
James

**Comment by m1ga** [comment]
Just use one of the existing plugins like https://wordpress.org/plugins/local-google-fonts/ or https://de.wordpress.org/plugins/custom-fonts/
You can even use Elementor Pro custom fonts (https://elementor.com/blog/custom-fonts/)

**Reply by ProfessorTechie** [reply]
While I understand that there are other plugins with partial functionality as what I described, I'm requesting to have this particular functionality incorporated into Elementor pro.

As I described above, I'm aware I can accomplish this by using Elementor's custom fonts. However, I would like to have such ability added to the WYSIWYG Editor.

Hence my post for a feature request. 

Using Elementor custom fonts for downloaded and locally hosted fonts that I can also select in the WYSIWYG editor (but not locally hosted), is both time consuming and somewhat annoying. 

If I find a Google font that I like while using Elementor's editor, I then have to go download it from Google's site, upload it to my site, and then select it as a custom font in Elementor.

I would prefer an option that when I select a Google font in the editor, I can chose to have Elementor add it locally to my site. Thus saving the time and energy I would normally to go thru. I feel the current options are quite roundabout  in their method. To host the fonts myself and then re-select the font I just selected in the editor, only this time as a custom font seems redundant.

And that's the joy of computers and programming, they can be adapted to improve efficiency and mental labor by having repetitive or redundant tasks automated.

Regards,
James

**Reply by m1ga** [reply]
Ok, I see. Since you only said to `host locally` and didn't mention Elementor Pro Custom fonts I wasn't sure if you know the Elementor Pro part :+1:  BTW: I've never had to chose a font using the editor since the fonts are normally chosen beforehand. 

If you look at the PRs you can find https://github.com/elementor/elementor/pull/17835 which sounds promising but is very old already and still experimental. But it looks like they are aware of it and working on it. There are some feature requests in the issue section about it too.

**Comment by nicholaszein** [comment]
This Feature Request was mentioned or is related to the following issues and discussions:

- #4544
- #23932
- #21716
- #28508

**Comment by toineenzo** [comment]
Maybe this PR can get picked up again? 👀
https://github.com/elementor/elementor/pull/17835

**Comment by toineenzo** [comment]
It just got merged!

https://github.com/elementor/elementor/pull/29732

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that Load Google Fonts Locally is now a native feature of Elementor since the release of version 3.27.0! 🥳 

You can now activate the new experimental feature by going to your WordPress **`Dashboard`** > **`Elementor`** > **`Settings`** > **`Features`** page, and switching the option to **`Load Google Fonts locally`** to **`Active`**.

This will make Elementor automatically download all the weights and style files of the Google Fonts in use in the **`.woff2`**, and save them in a special folder inside the **`../wp-content/uploads/`** folder.

These font files are downloaded only once a page using the Google Fonts is loaded, and will be re-downloaded on every new visit after the **`Elementor CSS & Data`** is **`Regenerated`**. Fonts are only loaded when in use.

For more information see: https://elementor.com/help/load-google-fonts-locally/

> _In the future this feature will be merged to core and may or may not have a setting in the **`Elementor Settings`** > **`Performance`** tab._

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
